### PR TITLE
ShipIt 54: Add-on theming updates

### PR DESCRIPTION
**Pull request by luke-underwood**
## These changes are part of a ShipIt 54 project SHPLIV-169.
**Theming adjustments to add-on to align styles closer with Atlassian Design System guidelines. This makes the add-on look like it belongs in Bitbucket.**

- Replaced UI with Atlassian Design System components
- Installed Storybook to improve add-on development experience. [Demo video here](https://www.loom.com/share/8a0b8ce13c2d4c77938a81061ab64405). Start using `yarn run storybook`
- Splits class component App.tsx into several smaller functional components. The separation makes it much easier to test in isolation, as Bitbucket is not required to develop the UI of the add-on
- Added loading states when performing actions "land" and "land when able". This provides feedback to the user that something is happening once they've clicked a button
- Added loading spinner while checking if able to land
- Fixed height to 200px with a scrollbar for overflow. This prevents issues caused by slow load times. Unfortunately Landkid's `/can-land` is usually the last thing to load on pull requests with times around 15 seconds. Transitioning from the loading state altered the height of the add-on, pushing the rest of the pull request downwards which causes frustrating cumulative layout shift. The fixed height prevents this happening. This is not an ideal solution but I think it's necessary unless we can improve response times from the API
- Error wording updates

### Before
landkid-before

### After
landkid-light-mode

A bonus of using Atlassian Design System components: It's Dark mode ready (won't be usable until supported in Bitbucket)

landkid-dark-mode

### CODE REVIEW

File: package.json
Comment (Blasz) [inline package.json:63]: Are the babel dependencies required for storybook?

File: src/static/bitbucket/components/Message.tsx
Comment (Blasz) [inline src/static/bitbucket/components/Message.tsx:108]: I understand this is to prevent CLS but 200 might not be tall enough. I can see quite a bit of text cut off in some of the storybook examples.

I'd like to see what this looks like in bitbucket itself.

**Review by Blasz — COMMENTED** [review]
Hey, style changes look great, thanks for contributing.

It'd be good to test this in our atlassian-frontend-landkid instance first before merging this. You'll need to re-push your branch in this `atlassian/landkid` repo itself rather than a fork so that it kicks off the build that publishes the docker image. From there you can consume the docker image in our deployment repo. You can visit go/oss to get access to our org.

**Comment by pastelsky** [comment]
I was just thinking of doing something similar and came across this PR. I can't help notice how out of place the current landkid block looks in PRs. Do you plan on taking this further @luke-underwood ?

**Comment by Brianzchen** [comment]
> It'd be good to test this in our atlassian-frontend-landkid instance first before merging this. You'll need to re-push your branch in this `atlassian/landkid` repo itself rather than a fork so that it kicks off the build that publishes the docker image. From there you can consume the docker image in our deployment repo. You can visit go/oss to get access to our org.

This is wrong now that I've migrated the docker image to github actions. @luke-underwood will need to merge this before it can be published

**Comment by Blasz** [comment]
We can't test landkid changes anymore without merging to master? That's concerning.

**Comment by Brianzchen** [comment]
> We can't test landkid changes anymore without merging to master? That's concerning.

I've just checked, it's not possible atm but build eng is suggesting they can implement a way to generate non-compliant tokens that can go into a different registry

**Comment by luke-underwood** [comment]
> I was just thinking of doing something similar and came across this PR. I can't help notice how out of place the current landkid block looks in PRs 😅. Do you plan on taking this further @luke-underwood ?

@pastelsky Yes still looking to finish this PR when I get time 😄

> This is wrong now that I've migrated the docker image to github actions. @luke-underwood will need to merge this before it can be published

@Brianzchen Got it. Can I merge this now or does it need approval first?

**Comment by pastelsky** [comment]
@Brianzchen Is there anyone else in the team who can sign this off? If the rollback time is fast enough, we can just go through with it at a time when queue is idle in our internal instance to test it.

File: src/routes/bitbucket/proxy/index.ts
Comment (Brianzchen) [inline src/routes/bitbucket/proxy/index.ts:52]: Getting serious now are we

**Review by Brianzchen — APPROVED** [review]
I'm ok with this change. It's very stylistic and at worst we can roll back this PR pretty easily. @luke-underwood do you plan to make additional changes? Otherwise we can merge and I can integrate it tomorrow

File: src/routes/bitbucket/proxy/index.ts
Comment (luke-underwood) [reply src/routes/bitbucket/proxy/index.ts:52]: 🫡 Too serious? I can only handle so many star wars references in my life

**Comment by luke-underwood** [comment]
@Brianzchen easy, thanks! I'll merge now then.

I'll be available tomorrow if anything goes wrong. Without testing my changes I'd say there's a good chance I'll need to make some additional changes, but if we can rollback easily it shouldn't be an issue

**Comment by pastelsky** [comment]
Looks like the build is red now due to a bunch of new TS errors —
https://app.circleci.com/pipelines/github/atlassian/landkid/509/workflows/8f53aaac-d6d0-4a6b-8d08-0ab3125a3bec/jobs/1816

**Pull request merged**
